.github/workflows/ci.yml

```
@@ -158,97 +181,118 @@ jobs:
+    timeout-minutes: 15  # Add job-level timeout
     needs: [build-matrix]
-    
+
     steps:
       - name: Download all artifacts
         uses: actions/download-artifact@v4
-      
+
       - name: Install ARM toolchain
+        timeout-minutes: 5  # Explicit timeout for apt operations
         run: |
```

actionlint (1.7.10)

[error] shellcheck reported issue in this script: SC2086:info:14:8: Double quote to prevent globbing and word splitting

(shellcheck)